mono/metadata/class.c

```diff
@@ -52,6 +52,7 @@
 #include <mono/utils/bsearch.h>
 #include <mono/utils/checked-build.h>
 
+/* Use mono_stats_lock and mono_stats_unlock when reading/writing stats. */
```

Comment:
This understates what the lock is intended to do. It should protect `mono_stats`, `mono_jit_stats`, and `mono_stats_method_desc` when you're not using atomics or have to read/modify two values that are dependent on each other (the simple example being the largest method size and name). I think, given those constraints, you're still missing some places where we should be taking it.